crates/saya/core/src/lib.rs

```diff
@@ -196,30 +198,39 @@ impl Saya {
             state_updates: state_updates_to_prove,
         };
 
-        println!("Program input: {}", new_program_input.serialize()?);
-
-        // let to_prove = ProvedStateDiff {
-        //     genesis_state_hash,
-        //     prev_state_hash: prev_block.header.header.state_root,
-        //     state_updates: state_updates_to_prove,
-        // };
+        let world_da = new_program_input.da_as_calldata(self.config.world_address);
+        let world_da_printable: Vec<String> = world_da.iter().map(|x| x.to_string()).collect();
+        trace!(target: "saya_core", "World DA {world_da_printable:?}.");
 
         trace!(target: "saya_core", "Proving block {block_number}.");
-        let proof = prover::prove(new_program_input.serialize()?, self.config.prover).await?;
+        let proof =
+            prove_stone(new_program_input.serialize(Some(self.config.world_address))?).await?;
         info!(target: "saya_core", block_number, "Block proven.");
 
-        // save proof to file
-        tokio::fs::File::create(format!("proof_{}.json", block_number))
-            .await
-            .unwrap()
-            .write_all(proof.as_bytes())
-            .await
-            .unwrap();
-
         trace!(target: "saya_core", "Verifying block {block_number}.");
-        let transaction_hash = verifier::verify(proof, self.config.verifier).await?;
+        let serialized_proof = parse_proof(&proof).unwrap();
+        let transaction_hash = verifier::starknet::starknet_verify(
+            self.config.fact_registry_address,
+            serialized_proof,
+        )
+        .await?;
         info!(target: "saya_core", block_number, transaction_hash, "Block verified.");
 
+        let ExtractProgramResult { program: _, program_hash } = extract_program(&proof)?;
+        let ExtractOutputResult { program_output: _, program_output_hash } =
+            extract_output(&proof)?;
+        let expected_fact = poseidon_hash_many(&[program_hash, program_output_hash]).to_string();
+        info!(target: "saya_core", expected_fact, "Expected fact.");
+
+        sleep(Duration::from_millis(5000));
+
+        trace!(target: "saya_core", "Applying diffs {block_number}.");
```

Comment:
As much as we can, we should use structured logging.
```suggestion
        trace!(target: LOG_TARGET, block_number, "Applying diff.");
```